doc/source/overview/config_files.md

```diff
@@ -0,0 +1,753 @@
+# Detector Configuration Files
+
+For advanced use cases, Alibi Detect features powerful configuration file based functionality. As shown below,
+**Drift detectors** can be specified with a configuration file named `config.toml` (adversarial and outlier 
+detectors coming soon!), which can then be passed to `load_detector`:
+
+
+`````{grid} 2
+
+````{grid-item-card}
+:shadow: md
+:margin: 1
+:padding: 1
+:columns: auto
+
+
+**Standard instantiation**
+^^^
+
+```python
+import numpy as np
+from alibi_detect.cd import MMDDrift
+
+x_ref = np.load('detector_directory/x_ref.npy')
+detector = MMDDrift(x_ref, p_val=0.05)
+```
+````
+
+````{grid-item-card}
+:shadow: md
+:margin: 1
+:padding: 1
+:columns: auto
+
+**Config-driven instantiation**
+^^^
+
+<p class="codeblock-label">config.toml</p>
+
+```{code-block} toml
+
+name = "MMDDrift"
+x_ref = "x_ref.npy"
+p_val = 0.05
+```
+
+```python
+from alibi_detect.utils.loading import load_detector
+filepath = 'detector_directory/'
+detector = load_detector(filepath)
+```
+````
+`````
+
+Compared to *standard instantiation*, config-driven instantiation has a number of advantages:
+
+- **Human readable**: The `config.toml` files are human-readable, providing a readily accessible record of 
+previously created detectors.
+- **Flexible artefact specification**: Artefacts such as datasets and models can be specified as locally serialized
+objects, or as runtime registered objects (see [Specifying complex fields](complex_fields)). Multiple detectors can 
+share the same artefacts, and they can be easily swapped.
```

Comment:
Can we claim this already in practice? I feel like there would be some relative path and/or symlink shenanigans...

Reply:
I had in mind that absolute file paths would be used in this case. I did originally implement to allow absolute paths but admittedly haven't tested recently. I will test again if we decide we want to retain this capability (otherwise will remove mentions in the docs).

Reply:
I've added a test `test_load_absolute` to test that `load_detector` works with absolute paths in the `config.toml`.